csv/result.go

```diff
@@ -1271,3 +1305,25 @@ func NewMultiResultEncoder(c ResultEncoderConfig) flux.MultiResultEncoder {
 		Encoder:   NewResultEncoder(c),
 	}
 }
+
+// bufferedCSVReader allows for unreading a single line of the csv data
+type bufferedCSVReader struct {
+	r    *csv.Reader
+	line []string
+}
+
+// Read returns the next line in the csv stream
+func (b *bufferedCSVReader) Read() ([]string, error) {
+	if len(b.line) > 0 {
+		line := b.line
+		b.line = nil
+		return line, nil
+	}
+	return b.r.Read()
+}
+
+// Unread places the provided line back on the buffer.
+// It is invalid to call unread multiple times without calling Read inbetween.
+func (b *bufferedCSVReader) Unread(line []string) {
+	b.line = line
```

Comment:
I wonder if there should be any guards here to avoid the perception that clients could unread several lines and expect them all to be buffered. Perhaps a second call to `Unread` when `b.line` is already set should produce an error?

Reply:
I like that idea, i'll update to return and error and we can check for it.